src/main/scala/scalismo/mesh/surfaceDistance/SurfaceDistance.scala

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright 2015 University of Basel, Graphics and Vision Research Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package scalismo.mesh.surfaceDistance
+
+import breeze.numerics.pow
+import scalismo.geometry.{Dim, Point, Vector, _3D}
+import scalismo.mesh.{TriangleId, TriangleMesh3D}
+import scalismo.mesh.surfaceDistance.BSDistance._
+
+
+/**
+  * Type of the closest point. At the moment the names are only suited for a triangular mesh.
+  */
+object ClosestPointType extends Enumeration {
+  type ClosestPointType = Value
+  val POINT, ON_LINE, IN_TRIANGLE = Value
+}
+import ClosestPointType._
+
+/**
+  * Descritpion of a closest point
+  * @param distance2 the squared distance
+  * @param pt the coordinates
+  * @param ptType on which geometric entity of the surface the closest point lies
+  * @param bc the barycentric coordinates of the point. The interpretation depends on the ptType.
+  * @param idx the index in the original surface instance of the geometric entity where the closest point lies. The interpretation depends on the ptType.
+  */
+case class ClosestPoint(val distance2: Double,
+                        pt: Vector[_3D],
+                        ptType: ClosestPointType,
```

Comment:
This is a bit too C-ish way of typing  :)  .. If these points really need to be differentiated then why not create real types?